### ISSUE TITLE
Message mismatch, possible encoding issue

### ISSUE THREAD

**Issue by andresfelipe-mejia** [ibm-messaging/kafka-connect-mq-source#85]
Hi!

I've been working with this connector for about 6 months and I have an issue with a message received from an IBM MQ Queue.

Tracing the Queue's  log, the message stored on the Queue is as follows:

```
 16:32:57.194679     4026.6      ZSESS:000003        0x01c0:  00000000 ffffffff **c4d6e3c2 c3e3d6d5**  |................|
 16:32:57.194679     4026.6      ZSESS:000003        0x01d0:  **f0f0f0f1 f0f0f8f3 f7f8f8f1 f0f0f2c1**  |................|
 16:32:57.194679     4026.6      ZSESS:000003        0x01e0:  **d9d70000 00000025 14919f00 00000000**  |.......%........|
 16:32:57.194679     4026.6      ZSESS:000003        0x01f0:  **0000000f 00000000 00000000 0f000000**  |................|
 16:32:57.194679     4026.6      ZSESS:000003        0x0200:  **00000000 000f0000 00000000 00000f00**  |................|
 16:32:57.194679     4026.6      ZSESS:000003        0x0210:  **00000000 0000000f 00000000 00000000**  |................|
 16:32:57.194679     4026.6      ZSESS:000003        0x0220:  **0f000000 00000000 000f0000 00000025**  |...............%|
 16:32:57.194679     4026.6      ZSESS:000003        0x0230:  **30100f00 00000000 0000000f f2f0f2f1**  |0...............|
 16:32:57.194679     4026.6      ZSESS:000003        0x0240:  **f0f3f2f5 00000000 00000000 0f000000**  |................|
 16:32:57.194679     4026.6      ZSESS:000003        0x0250:  **00000000 000f0000 00000000 00000f00**  |................|
 16:32:57.194679     4026.6      ZSESS:000003        0x0260:  **00000000 0000000f f0f0f0f0 f0f0f0f0**  |................|
 16:32:57.194679     4026.6      ZSESS:000003        0x0270:  **00000000 00000000 0f000000 00000000**  |................|
 16:32:57.194679     4026.6      ZSESS:000003        0x0280:  **000f0000 00000000 00000f00 00000000**  |................|
 16:32:57.194679     4026.6      ZSESS:000003        0x0290:  **0000000f 00000000 00000000 0f000000**  |................|
 16:32:57.194679     4026.6      ZSESS:000003        0x02a0:  **00000000 000f0000 00000000 00000f00**  |................|
 16:32:57.194679     4026.6      ZSESS:000003        0x02b0:  **00000000 0000000f 00000000 00000000**  |................|
 16:32:57.194679     4026.6      ZSESS:000003        0x02c0:  **0f000000 00000000 000f0000 00000000**  |................|
 16:32:57.194679     4026.6      ZSESS:000003        0x02d0:  **00000f00 00000000 0000000f 40404040**  |............@@@@|
 16:32:57.194679     4026.6      ZSESS:000003        0x02e0:  **40404040 00000000 00000000 0f000000**  |@@@@............|
 16:32:57.194679     4026.6      ZSESS:000003        0x02f0:  **00000000 000f0000 00000000 00000f00**  |................|
 16:32:57.194679     4026.6      ZSESS:000003        0x0300:  **00000000 0000000f f0f0f0f0 f0f0f0f0**  |................|
 16:32:57.194679     4026.6      ZSESS:000003        0x0310:  **00000000 00000000 0f000000 00000000**  |................|
 16:32:57.194679     4026.6      ZSESS:000003        0x0320:  **000f0000 00000200 00000ff2 f0f2f0f0**  |................|
 16:32:57.194679     4026.6      ZSESS:000003        0x0330:  **f9f0f3f1**                             |....            |
```

In **bold** I highlighted the payload of the message.

But, when the connector consumes the message is received like this:

```
c4d6e3c2 c3e3d6d5 f0f0f0f1 f0f0f8f3 
f7f8f8f1 f0f0f2c1 d9d70000 00000015
14919f00 00000000 0000000f 00000000
00000000 0f000000 00000000 000f0000
00000000 00000f00 00000000 0000000f
00000000 00000000 0f000000 00000000
000f0000 00000015 30100f00 00000000
0000000f f2f0f2f1 f0f3f2f5 00000000
00000000 0f000000 00000000 000f0000
00000000 00000f00 00000000 0000000f
f0f0f0f0 f0f0f0f0 00000000 00000000
0f000000 00000000 000f0000 00000000
00000f00 00000000 0000000f 00000000
00000000 0f000000 00000000 000f0000
00000000 00000f00 00000000 0000000f
00000000 00000000 0f000000 00000000
000f0000 00000000 00000f00 00000000
0000000f 40404040 40404040 00000000
00000000 0f000000 00000000 000f0000
00000000 00000f00 00000000 0000000f
f0f0f0f0 f0f0f0f0 00000000 00000000
0f000000 00000000 000f0000 00000200
00000ff2 f0f2f0f0 f9f0f3f1
```

Notice that every group stored in the Queue as 00000025 are later read as 00000015.

In the configuration we are using a String value converter.

In another java application we are consuming the topic with a String SerDe and then getting the byte array using Cp037 code page.

We have confirmed that the Queue is using that Encoding.

Any help would be useful.

Thanks in advance

**Comment by jay999123** [comment]
Try convert the data that you receive to string format from your java application - 

`new String(bytedata, "<MQ schema>");`

may be `new String(bytedata, "Cp037");`

**Comment by andresfelipe-mejia** [comment]
Hi, thank you for your help.

I think I have understood my problem.

Apparently, the connector, as configured, was doing a String conversion with UTF-8 encoding as default, since I don't know how to change that behavior, I modified the connector to not use a String converter but a ByteArray converter, therefore, not modifying the message. Then, in my Java application, I used your suggestion and now I can see the 00000025 fields.

Right now I'm testing whether another values have issues or not. 

Hope that helps anyone and again, thanks for the tip.
